### PR TITLE
Add support for collision layers and masks in CSG shapes

### PR DESCRIPTION
**Pull request by ghost**
Code mostly copied from PhysicsBody. Not having this hampers the usability of CSG shapes, so it might make sense to have it for 3.1. Not sure if the "use_collision" flag should be removed as a result of this.

### CODE REVIEW

File: modules/csg/csg_shape.cpp [hunk 1]
```diff
@@ -381,7 +441,7 @@ void CSGShape::_notification(int p_what) {
 			parent->_make_dirty();
 		parent = NULL;
 
-		if (use_collision && is_root_shape()) {
+		if (use_collision && is_root_shape() && root_collision_instance.is_valid()) {
```
Comment (cyclopsian) [inline modules/csg/csg_shape.cpp]: Fixes a separate issue with a NULL free error that could happen when a CSG shape with use_collision enabled was moved inside another shape:

```
ERROR: free: Invalid ID
   At: modules/bullet/bullet_physics_server.cpp:1536
```

**Comment by akien-mga** [comment]
@reduz mentions it should be fine to merge, could you rebase to fix the conflicts?

**Comment by akien-mga** [comment]
I pushed a rebased commit myself, will merge once CI tests are OK.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**